akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala

```diff
@@ -243,17 +271,21 @@ import akka.annotation.InternalApi
 private[typed] class GuardianActorAdapter[T](_initialBehavior: Behavior[T]) extends ActorAdapter[T](_initialBehavior) {
   import Behavior._
 
+  def receive: Receive = waitingForStart(Nil)
+
   override def preStart(): Unit =
     if (!isAlive(behavior))
       context.stop(self)
-    else
-      context.become(waitingForStart(Nil))
 
   def waitingForStart(stashed: List[Any]): Receive = {
     case GuardianActorAdapter.Start =>
       start()
-
-      stashed.reverse.foreach(receive)
+      if (Behavior.isAlive(behavior)) {
```

Comment:
why is the alive check needed here? It was checked in `preStart` and will not change as far as I can see. if it really can change we should probably stop if not alive

Reply:
`start` starts any deferred behavior it so it could be stopped when we get here.